src/vos/vos_dtx.c

```diff
@@ -1126,20 +1147,28 @@ vos_dtx_check_availability(daos_handle_t coh, uint32_t entry,
 
 	found = lrua_lookupx(cont->vc_dtx_array, entry - DTX_LID_RESERVED,
 			     epoch, &dae);
+	if (!found) {
+		D_DEBUG(DB_TRACE,
+			"Entry %d "DF_U64" not in lru array, it must be committed\n",
+			entry, epoch);
 
-	/* The DTX owner can always see the DTX. */
-	if (found && dtx_is_valid_handle(dth) && dae == dth->dth_ent)
 		return ALB_AVAILABLE_CLEAN;
+	}
+
+	if (intent == DAOS_INTENT_PURGE) {
+		/*
+		 * The DTX entry still references related data record,
+		 * then we cannot (vos) aggregate related data record.
+		 */
+		D_WARN("DTX "DF_DTI" (%u) still references the data, cannot be (VOS) aggregated\n",
+		       DP_DTI(&DAE_XID(dae)), vos_dtx_status(dae));
```

Comment:
Is this a temporary debug patch or it's targeted for landing? BTW, is it related to DAOS-12260?

Reply:
It is not temporary debug patch, instead, it will be landed to master branch. That will not cause huge logs because it is only used for VOS aggregation and relative rare to hit.

I am not suer whether it is related with DAOS-12260 or not, I do not know the operation sequence at that time for this ticket. The direct reason of this issue is that VOS aggregation hit non-committed DTX entries for long time. It is necessary to log related DTX status under such case to well understand what happened.

Reply:
why is it a problem if we attempt to aggregate an in-progress entry?  As long as we don't allow it, it should be fine, no?  Sorry, apparently this message is showing frequently in testing.